rust/src/lib.rs

```diff
@@ -2177,16 +2215,41 @@ pub extern "C" fn zcashlc_shield_funds(
             .cloned()
             .collect();
 
-        shield_transparent_funds(
-            &mut update_ops,
-            &network,
-            LocalTxProver::new(spend_params, output_params),
-            &usk,
-            &taddrs,
-            &memo_bytes,
-            ANCHOR_OFFSET,
-        )
-        .map_err(|e| format_err!("Error while shielding transaction: {}", e))
+        if use_zip317_fees {
+            let input_selector = GreedyInputSelector::new(
```

Comment:
can this be refactored to 
````
let input_selector  = if  use_zip317_fees {
 // create ZIP-317 selector
 } else {
  // create ZIP-313 selector
  }
  
// single call to 
   shield_transparent_funds(
                &mut update_ops,
                &network,
                LocalTxProver::new(spend_params, output_params),
                &input_selector,
                &usk,
                &taddrs,
                &memo_bytes,
                ANCHOR_OFFSET,
            )
            .map_err(|e| format_err!("Error while shielding transaction: {}", e))
            
````

Reply:
Sadly no, because the two branches of the conditional have different types. In the equivalent PR on the Android SDK, I squared the circle (and also deduplicated the logic between `createToAddress` and `shieldToAddress`) with this helper: https://github.com/zcash/zcash-android-wallet-sdk/pull/775/files#diff-ea14a0f1195cb3440fbdf0cfba825a92078a0d1ff9655232f182000d360e2976R1069